dsgrid/config/dimension_associations.py

```diff
@@ -108,7 +96,7 @@ def get_associations(self, *dimensions, data_source=None):
         ds_column = DimensionType.DATA_SOURCE.value
         columns = [ds_column] + [x.value for x in dimensions]
         table = self._table.select(*columns)
-        if data_source is not None:
+        if data_source is not None and ds_column in table.columns:
```

Comment:
Wouldn't it be an error if data_source is specified but there is no "data_source" in the table?

Reply:
It would return the entire table down-selected to the specified dimensions, isn't it? If data_source is not specified in any paired association, we assume the table applies to all data_sources, right?